### PR TITLE
Update/Use the new Terms of Service package in Jetpack

### PR DESCRIPTION
**Pull request by enejb**
This PR deprecated the Jetpack ::jetpack_tos_agreed().
Updates any calls that relied on that method the the new Terms_of_service->has_agreed();

Also instantiates any Jetpack tracking as soon as the sites agrees to the TOS.

Supersedes #13759

#### Changes proposed in this Pull Request:
* Fixes tracking of new connections, integrates the new Terms Of Service Package in Jetpack.

#### Testing instructions:
* Connect Jetpack, does it work as expected. 
* Does tracking for the connected user work as expected?
* Do you see the jetpack_connection_success again? After the site is connected.

#### Proposed changelog entry for your changes:
* Integrates the new Terms Of Service Package in Jetpack

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -418,19 +441,62 @@
...
+                "url": "https://github.com/Automattic/jetpack-tracking.git",
+                "reference": "c3ef8d03a1c6b64a008b2a33b10c504b069a4800"
+            },
             "dist": {
-                "type": "path",
-                "url": "./packages/tracking",
-                "reference": "b2c79c42a8ccd728db9450818aa95fd6d51afc85",
-                "shasum": null
+                "type": "zip",
+                "url": "https://api.github.com/repos/Automattic/jetpack-tracking/zipball/c3ef8d03a1c6b64a008b2a33b10c504b069a4800",
```
Comment (tyxla) [inline composer.lock]: We should revert those changes so we'll be able to pull from the local version of the package and take advantage of the changes that this PR is introducing.

Comment (tyxla) [reply composer.lock]: Note that you will still need the changes to `composer.lock` that introduce the ToS package as a dependency.

**Review by jeherve — DISMISSED** [review]
All tests appear to be failing right now.

**Comment by tyxla** [comment]
Updated, tests should be fixed now. We just had a little typo and an needed to update `composer.lock`.

File: class.jetpack.php [hunk 2]
```diff
@@ -727,6 +724,22 @@ function( $callable ) {
 		 */
 		add_action( 'script_loader_tag', array( $this, 'script_add_async' ), 10, 3 );
 	}
+	/**
+	 * Runs after all the plugins have loaded but before init.
+	 */
+	function after_plugins_loaded() {
+
+		$terms_of_service = new Terms_Of_Service();
+		$tracking = new \Automattic\Jetpack\Plugin\Tracking();
```
Comment (lezama) [inline class.jetpack.php]:
```suggestion
		$tracking = new Tracking();
```

**Review by lezama — DISMISSED** [review]

**Review by delawski — APPROVED** [review]
The code looks good to me. I've tested it locally and the connection flow worked for me as expected. I also found an event I triggered in https://mc.a8c.com/tracks/live/

**Pull request merged**